pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1224,6 +1244,18 @@ public LLRealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableCo
       Schema schema, LLCSegmentName llcSegmentName, Semaphore partitionGroupConsumerSemaphore,
       ServerMetrics serverMetrics, @Nullable PartitionUpsertMetadataManager partitionUpsertMetadataManager,
       @Nullable PartitionDedupMetadataManager partitionDedupMetadataManager) {
+      this(segmentZKMetadata, tableConfig, realtimeTableDataManager, resourceDataDir, indexLoadingConfig, schema,
+          llcSegmentName, partitionGroupConsumerSemaphore, serverMetrics, partitionUpsertMetadataManager,
+          partitionDedupMetadataManager, Clock.systemUTC());
+  }
+
+  @VisibleForTesting
+  protected LLRealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableConfig tableConfig,
```

Comment:
We don't need to introduce the `Clock` into this main production class. The `FakeLLRealtimeSegmentDataManager` is already overriding the `now()` method, and we may introduce the clock there